### DISCUSSION TITLE
<title> Github corrupts mms.c </>

### DISCUSSION THREAD

**Discussion by plater** [community/community#128034] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

I attempted to view https://github.com/plater/scandrop.X/blob/master/mms.c and for some reason the file doesn't display and thinks the file is binary. I opened the local version and it appeared to have non printable characters. I cleaned this up and continued to work on the file then on success pushed the project to the github repository.
I use linux and microchip's mplabx based on netbeans. I opened the local file in an independent editor - kwrite and found no errors. I'll try deleting the file locally and then push to delete it online and then restore it tomorrow but I would really like to know what's wrong so it doesn't happen again.

**Comment by AVMiraga** [comment]
The file you are inquiring about is either private or the link is incorrect.

**Reply by plater** [reply]
Sorry it's for a payment system and private but I loaded the file into a hex editor and incised the garbage which was mostly 0x00 or NULLS how it happened I don't know I've never heard of a text editor that can enter NULL characters. I've attached the fixed file-mms.c and the renamed corrupted file mmsuncr.c. The corrupted file loads into editors without any complaints but because it has zeros embeded in it it won't display on github. The project was working last night and the build system doesn't find an error.
[mmsc.zip](https://github.com/user-attachments/files/15786033/mmsc.zip)

**Reply by AVMiraga** [reply]
I don't think that GitHub corrupts the file, as Git verifies file integrity. It's more likely that the corruption occurred before it was pushed. Git calculates a checksum for the file and stores it along with the file. When the file is later retrieved, Git recalculates the checksum and compares it with the stored value to ensure the file has not been corrupted. File Corrupted possibly because of power loss or faulty disk.

**Reply by plater** [reply, marked as answer]
I think it happened when my keyboard died while working on the file. Would be nice for a binary file with a c extension threw a warning
